pkg/devfile/adapters/common/generic.go

```diff
@@ -140,13 +141,34 @@ func (a GenericAdapter) ExecDevfile(commandsMap PushCommandsMap, componentExists
 		// if we need to restart, issue supervisor command to stop all running commands first
 		if componentExists && IsRestartRequired(command) {
 			klog.V(4).Infof("restart:true, restarting %s", defaultCmd)
-			if cmd, err := newSupervisorStopCommand(command, a); cmd != nil {
+			if cmd, err := newSupervisorStopCommand(command, "", a); cmd != nil {
 				if err != nil {
 					return err
 				}
 				commands = append(commands, cmd)
 			}
 		} else {
+			// if there is switch of commands from run<->, we need to stop the other command
+			var otherCommand common.DevfileCommand
```

Comment:
I think we can just check against the env's `runMode` and detect the switch between the commands.

Reply:
What if the user tweaked the env file not knowing what they are doing?

Reply:
We don't recommend the user to tweak odo related files. Even if the user changes the namespace to a non existent namespace, it will fail too. If the user is tweaking the env file, the failure his/her responsibility.

Reply:
@mik-dass thanks for your suggestion, updated.